homeassistant/components/transmission/__init__.py

```diff
@@ -50,20 +53,21 @@
         vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
         vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
         vol.Optional(TURTLE_MODE, default=False): cv.boolean,
+        vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL):
+            cv.time_period,
         vol.Optional(CONF_MONITORED_CONDITIONS, default=['current_status']):
         vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
     })
 }, extra=vol.ALLOW_EXTRA)
 
-SCAN_INTERVAL = timedelta(minutes=2)
-
 
 def setup(hass, config):
     """Set up the Transmission Component."""
     host = config[DOMAIN][CONF_HOST]
     username = config[DOMAIN][CONF_USERNAME]
```

Comment:
Username and password are optional, so we can't use `dict[key]` for those. We have to use `dict.get`.